File: wisefy/src/androidTest/java/com/isupatches/wisefy/IsWifiEnabledTests.java
package com.isupatches.wisefy;


import com.isupatches.wisefy.base.TestActivity;
import org.junit.Test;
import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.when;


public class IsWifiEnabledTests extends BaseTestClass<TestActivity> {

    public IsWifiEnabledTests() {
        super(TestActivity.class);
    }

    @Test
    public void failure() {
        when(mMockWiFiManager.isWifiEnabled()).thenReturn(false);
        assertEquals(false, mWiseFy.isWifiEnabled());
    }

    @Test
    public void failure_missingPrerequisite() {
        missingPrerequisite();
        assertEquals(false, mWiseFy.isWifiEnabled());
    }

    @Test
    public void success() {
        when(mMockWiFiManager.isWifiEnabled()).thenReturn(true);
        assertEquals(true, mWiseFy.isWifiEnabled());
    }
}

File: wisefy/src/androidTest/java/com/isupatches/wisefy/IsWifiEnabledTests.kt
package com.isupatches.wisefy

import com.isupatches.wisefy.internal.base.BaseInstrumentationTest

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Tests the ability to determine if wifi is enabled on a device.
 *
 * @author Patches
 * @since 3.0
 */
internal class IsWifiEnabledTests : BaseInstrumentationTest() {

    @Test fun failure_prechecks() {
        mockWiseFyPrechecksUtil.isWifiEnabled_failure()
        assertEquals(false, wisefy.isWifiEnabled())
    }

    @Test fun failure() {
        mockWiseFyPrechecksUtil.isWifiEnabled_success()
        mockNetworkUtil.isWifiEnabled(false)
        assertEquals(false, wisefy.isWifiEnabled())
    }

    @Test fun success() {
        mockWiseFyPrechecksUtil.isWifiEnabled_success()
        mockNetworkUtil.isWifiEnabled(true)
        assertEquals(true, wisefy.isWifiEnabled())
    }
}
